### PR TITLE
Made watch more intelligent by only rebuilding js or sass in build an…

### PR DESCRIPTION
**Pull request by AlbertoElias**
…d demo when one of those type of files has changed

I'm not sure the extra code hassle is worth it, or if this is the best way to do it. What do you think @samgiles ? It's definitely faster

### CODE REVIEW

**Comment by AlbertoElias** [comment]
Closes #259

File: lib/tasks/verify.js [hunk 1]
```diff
@@ -177,5 +177,5 @@ module.exports.origamiJson = function() {
 	return result;
 };
 
-module.exports.watchable = true;
+module.exports.watchable = false;
```
Comment (samgiles) [inline lib/tasks/verify.js]: Why change this to false?

Comment (AlbertoElias) [inline lib/tasks/verify.js]: It's not really useful as the watching stops as soon as there's a linting
error. I implemented the cleverer watch there, and then figured out that
verify --watch is just not useful

-- 

------------------------------
This email was sent by a company owned by Pearson plc, registered office at 
80 Strand, London WC2R 0RL.  Registered in England and Wales with company 
number 53723.

**Comment by samgiles** [comment]
Yeh the extra code is annoying but with this architecture is the simplest thing to do. I think there's probably a cleaner way to do it though: it feels like 'watch' should be a trait of a task rather than a task itself.

Maybe you could invoke, for example, the `js` build task with a `watch` option and it task itself will register a `watch` on only the files that task is interested in for example.

This is something to consider I think, but for now this PR LGTM, especially as it makes watch quicker.

**Comment by AlbertoElias** [comment]
I thought about that, but I think that might end up requiring a lot of repeated boilerplate code

**Comment by samgiles** [comment]
imo boilerplate code ==  good code :D.   I honestly think it would be fine.

**Pull request merged**